corehq/tabs/uitab.py

```diff
@@ -127,7 +124,8 @@ def should_show(self):
         if not self.show_by_default and not self.is_active_tab:
             return False
 
-        if self.restrict_access_by_location and not self.filtered_dropdown_items:
```

Comment:
I'm not sure I follow the part about how it used to swallow the errors.  If a template references a dynamic property and that dynamic property raises an exception, the exception is still raised.

Could it be the case that this template never referenced the dropdown or sidebar items, but now something elsewhere in the python code is referencing them on this request?